round4 (#17)
ofirgall/learn-nvim: chapters/02-basic-config.md

`lvim.plugins` is passed to [packer](https://github.com/wbthomason/packer.nvim), `folke/tokyonight.nvim` is a short for [github.com/folke/tokyonight.nvim](https://github.com/folke/tokyonight.nvim).

Useful Packer Commands:
* `:PackerInstall` - Install new added plugins.
* `:PackerStatus` - List all installed pluigns.

> * `:PackerSnapshot` - Take snapshot of your current plugins versions (useful before updating).

Reviewer: ```suggestion
* `:PackerSnapshot` - Take a snapshot of your current plugins versions (useful before updating).
```

* `:PackerUpdate` - Update all plugins.

> * `:PackerClean` - Removed unused plugins.

Reviewer: ```suggestion
* `:PackerClean` - Remove unused plugins.
```

* [guess-indent.nvim](https://github.com/NMAC427/guess-indent.nvim) - Automatic indentation style detection.
* [nvim-autopairs](https://github.com/windwp/nvim-autopairs) - Auto pair, when typing a pair, eg: `(`, add the closure and set the cursor in between. (Included in LunarVim)

##### Programming languages support

> Some languages has plugins to support them specific, usually the plugin setup the LSP server (read below), make sure you don't setup the LSP more than once. I recommend to skip the LSP setup of the plugin and set it by yourself/mason.

Reviewer: ```suggestion
Some languages have plugins to support them specifically, usually the plugin takes care of the LSP setup (read below), so make sure you don't setup the LSP more than once. I recommend to skip the LSP setup of the plugin and set it up yourself or with mason.
```

E.g: [nvim-treesitter-textobjects](https://github.com/nvim-treesitter/nvim-treesitter-textobjects) which allows to do an action on a code context such as `copy` the `function/class/argument/loop` and more, we will elaborate on that later.

> _**Note:**_ make sure you to install treesitter parser for your language by adding it to the `ensure_installed` array or run `:TSUpdate {lang}`.

Reviewer: ```suggestion
_**Note:**_ make sure you to install treesitter parser for your language by adding it to the `ensure_installed` table or run `:TSUpdate {lang}`.
```
